release-workflow.md

```diff
@@ -0,0 +1,54 @@
+# Workflow for Releasing Different Versions
+
+We release the latest versions of the main repository including the content from _The Turing Way_ book on Zenodo (DOI for all versions: [10.5281/zenodo.3233853](https://doi.org/10.5281/zenodo.3233853)).
+In this document, we describe the process for making a release on Zenodo through GitHub.
+
+We don't describe how to connect a GitHub account/repository to a Zenodo account.
+To learn more about that, please read the details in [this post](https://docs.github.com/en/repositories/archiving-a-github-repository/referencing-and-citing-content).
+
+## Updating release information
+
+- Update CITATION.cff file:
+  - Update `version: 1.0.1` (currently in line 7), `date-released: "2021-11-10"` (currently in line 13)
+- Validate the file locally on your computer using `cffconvert` (here we assume that the user knows how to use their terminal)
+  - Step 1: Please install cffconvert using the following command: `python3 -m pip install --user cffconvert` (details here: https://pypi.org/project/cffconvert/)
+  - Run the following command: `cffconvert --validate`
+- Once validated, create .zenodo.json file using the following command: `cffconvert --format zenodo > .zenodo.json`
+
+Please note that we have currently left out individual names from the CITATION file, however, this is open for discussion if we will add all the authors manually or have some other automated ways to do that.
+
+## Drafting release on GitHub
+
+- Click on [the release option](https://github.com/alan-turing-institute/the-turing-way/releases) on GitHub main repository
+- Draft a [new release](https://github.com/alan-turing-institute/the-turing-way/releases/new)
+- Click 'Choose a tag', provide a new version name (such as v1.0.1, v1.02, ...) and click the 'create a new tag on release' option
```

Comment:
Is it worth adding a description of when would we consider a version to be [major, minor or patch](https://semver.org/) ?  I would imagine something like:
 - patch: Small additions to chapters are patch (e.g. between book-dashes)
 - minor: Significant number of new content & new chapters (e.g. after a book dash)
 - major: Major changes (e.g. major re-arranging of chapters into different guides)

Reply:
The instructions look good to me.

Reply:
```suggestion
- Click 'Choose a tag', provide a new version name (such as v1.0.1, v1.02, ...) and click the 'create a new tag on release' option
- A note on when we consider a version to be [major, minor or patch](https://semver.org/):
  - patch: Small additions to chapters are patch such as bug fixing, editing or minor contributions between the Book Dashes
  - minor: Significant number of new content & new chapters such as during a Book Dash
  - major: Major changes such as major re-arranging of chapters into different guides or addition of a new guide
```